### ISSUE TITLE
Enhance the Empty filter for Distance to filter "Unknown" aka "NaN" values

### ISSUE THREAD

**Issue by BrettEBowman** [fratsloos/fr24_card#14]
For some of the flights in my list, the location is apparently "Unknown" or "NaN" in HA terms. The fr24_card calculation for Distance can't calculate it since the location is unknown, so it carries over to the Distance value as "NaN" also.

I would like to NOT display those flights with an unknown distance. I tried using the
`hide:
  empty:
    - distance`
configuration, but apparently since the Distance field has something in it (the "NaN"), they aren't filtered.

**Comment by fratsloos** [comment]
@BrettEBowman Can you supply me with the JSON for when you have that result?

**Comment by BrettEBowman** [comment]
Here's a .json file as well as what the fr24 card displays. (Both from approximately the same time, so they should roughly match up.)

[aircraft.zip](https://github.com/fratsloos/fr24_card/files/9174010/aircraft.zip)

**Comment by fratsloos** [comment]
Thanks, @BrettEBowman. I've tested it with this JSON, and I don't see the `NaN`. This might be because of the client. What browser or app are you using when you get the `NaN`?

Can you also give me YAML of the card confirguration?

**Comment by BrettEBowman** [comment]
I'm using the latest Goolge Chrome on Windows 10 for the browser.

Here's the YAML for the card:
```type: custom:fr24-card
entity: sensor.msp_piaware_flights
columns:
  - flag
  - registration
  - flight
  - icon
  - altitude
  - speed
  - distance
hide:
  empty:
    - distance
popup: true
zone: zone.home
sort: distance
```
I did look at the .json and it looked like flights that were showing up that way did have lat and long values. So maybe the problem has something to do with the math to determine the distance. The values for my zone.home are latitude: 44.97865681506032, longitude: -93.05218862841296.

Another possible clue that I've noticed is that when I leave that page displayed and it gets updated, the flights often are no longer sorted by distance. When I just opened the page for the very first time today to get the YAML for this post, I noticed that the first time, all of the flights had distance values. But then when it refreshed, a couple had the "NaN". It makes me wonder if there might be a problem with variable reinitialization or something like that.

**Comment by fratsloos** [comment]
Thanks, that is useful info. I'm looking into it. It might have something to do with the coordinates of the zone. When I used your card config I had no issue. When I added an extra zone with your coordinates I did get a `NaN`.

**Comment by fratsloos** [comment]
As mentioned I found the `NaN` in my test environment when I used your configuration and the first zone you gave. With the second zone I didn't have it. But that might be because of the content of `aircraft.json`. Same as I didn't get it with my local data.

I used a different method to calculate the distance. This time based on the [documentation of Google](https://cloud.google.com/blog/products/maps-platform/how-calculate-distances-map-maps-javascript-api). With his method I don't see the `NaN` with your data and configuration.

Before:
![before](https://user-images.githubusercontent.com/9197177/180853728-d76162da-cb20-4788-bde6-9fe4af76ba9d.png)

After:
![after](https://user-images.githubusercontent.com/9197177/180853763-d4832b5a-fcc3-43de-8c15-8bc22bcbc4e6.png)

Are you able to test this branch before I merge it into the code?
I didn't see an option to use anything other than `master` or a version in HACS. So if you want to test it you may have to download it. See the instructions on https://github.com/fratsloos/fr24_card/blob/master/readme/installation.md.

Just saw a typo on that page, will fix that with the next release :)

**Comment by BrettEBowman** [comment]
Sure, I can manually download and install to test. Should I use the develop branch or that specific bugfix branch?

**Comment by fratsloos** [comment]
You should use the [bugfix](https://github.com/fratsloos/fr24_card/tree/bugfix/14-enhance-the-empty-filter-for-distance-to-filter-unknown-aka-nan-values) branch connected to this issue.

**Comment by BrettEBowman** [comment]
It's possible that I didn't manually install properly... I copied the files from the dist folder onto my HA server in the folder where the fr24 files were. There weren't .gz versions in the downloaded dist folder, so I created those. I restarted the HA supervisor, opened my Flights dashboard, and did a forced Ctrl-F5 refresh.

At first, it seemed better, but then I started getting the NaNs again. I captured a couple of aircraft.json files and corresponding displays.

[aircraft1.zip](https://github.com/fratsloos/fr24_card/files/9184460/aircraft1.zip)

**Comment by BrettEBowman** [comment]
I think I figured out what I did wrong in the installation. I didn't change the Resource mapping, so it still had references to the HACS package. When I redid it, then the page started working OK. And I'm not seeing the NaNs. So, I think the change fixed the problem.

**Comment by fratsloos** [comment]
Thanks for testing. The compressed files (`.gz`) are created by HACS. I'll check the new JSON files in the coming days. Just to be sure, were you testing from the first or second zone?

**Comment by BrettEBowman** [comment]
That testing was from my primary home zone which was the first one.

**Comment by fratsloos** [comment]
> I think I figured out what I did wrong in the installation. I didn't change the Resource mapping, so it still had references to the HACS package. When I redid it, then the page started working OK. And I'm not seeing the NaNs. So, I think the change fixed the problem.

Great! :) Just to be sure I'll test with the new JSON files as well before I release the fix.

**Comment by fratsloos** [comment]
The fix is published in the latest release: [v0.5.1](https://github.com/fratsloos/fr24_card/releases/tag/v0.5.1).
